File: best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let lowest = prices[0];
    let bestProfit = 0;

    for (let current of prices) {
        if (lowest > current) lowest = current;
        bestProfit = (Math.max(bestProfit, current - lowest))
    }
    return bestProfit;
};


console.log(maxProfit([7,1,5,3,6,4])); //5
console.log(maxProfit([7,6,4,3,1])); //0

File: best-time-to-buy-and-sell-stock.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        lowest = prices[0]
        bestProfit = 0

        for current in prices:
            if lowest > current:
                lowest = current
            bestProfit = (max(bestProfit, current - lowest))
        return bestProfit

solution = Solution()

print(solution.maxProfit([7,1,5,3,6,4]))  #5
print(solution.maxProfit([7,6,4,3,1]))  #0